### ISSUE TITLE
IllegalArgumentException on LostApiClient.disconnect()

### ISSUE THREAD

**Issue by westnordost** [mapzen/lost#178]
### Description
I am getting the following crash reports from users of my app:

```
java.lang.IllegalArgumentException: Service not registered: com.mapzen.android.lost.internal.FusedLocationProviderApiImpl@96b2509
	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1289)
	at android.app.ContextImpl.unbindService(ContextImpl.java:1483)
	at android.content.ContextWrapper.unbindService(ContextWrapper.java:648)
	at com.mapzen.android.lost.internal.FusedLocationProviderApiImpl.onDisconnect(FusedLocationProviderApiImpl.java:55)
	at com.mapzen.android.lost.internal.FusedLocationServiceConnectionManager.disconnect(FusedLocationServiceConnectionManager.java:70)
	at com.mapzen.android.lost.internal.FusedLocationProviderApiImpl.disconnect(FusedLocationProviderApiImpl.java:92)
	at com.mapzen.android.lost.internal.LostApiClientImpl.disconnect(LostApiClientImpl.java:60)
	at de.westnordost.streetcomplete.data.QuestAutoSyncer.stopPositionTracking(QuestAutoSyncer.java:85)
	at de.westnordost.streetcomplete.data.QuestAutoSyncer.onStop(QuestAutoSyncer.java:69)
	at de.westnordost.streetcomplete.MainActivity.onStop(MainActivity.java:280)
	at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1289)
	at android.app.Activity.performStop(Activity.java:6854)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4160)
	... 9 more
```

### Steps to Reproduce
This is a crash report logged to the Android developer console. The code at that location looks like this:

```
if(lostApiClient.isConnected())
{
	LocationServices.FusedLocationApi.removeLocationUpdates(lostApiClient, this);
	lostApiClient.disconnect();
}
```

So, LOST complains that the service is not registered, even though I asked the client before if it is connected.

### Lost & Android Version
Nexus 5X (bullhead)
Android 7.1
LOST 2.1.2 (can't update to 2.2.0 because of #170 yet)

**Comment by ecgreb** [comment]
Thanks for the report. Its possible that if all location updates are removed the service is disconnected (incorrectly) but will need more investigation to confirm if that is the case. Stay tuned on this one.

**Comment by westnordost** [comment]
Got some more crash reports now. The crashes happen independently of the Android version.

**Comment by sarahlensing** [comment]
@westnordost Can you give the LOST `2.3.0-SNAPSHOT` a test? Testing locally, this crash was resolved [here](https://github.com/mapzen/lost/commit/f9d8b9ccbb857ccf9d3ea884a9898751ca93dd06)

**Comment by westnordost** [comment]
Well I haven't found a way to reliably reproduce it, the above crash report is just something I found that several people using the app have through the Google Developer Console.

**Comment by sarahlensing** [comment]
Okay. I'm going to close this issue as it is is currently resolved in the `2.3.0` snapshot. Please re-open a new issue if you find there are still problems after upgrading. We should be releasing a new version of LOST within a week or so. Thank you!
